samtranslator/validator/sam_schema/definitions/api.json

```diff
@@ -0,0 +1,789 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "additionalProperties": false,
+  "type": "object",
+  "definitions": {
+    "AWS::Serverless::Api": {
+      "if": {
+        "properties": {
+          "Type": {
+            "const": "AWS::Serverless::Api"
+          }
+        },
+        "required": [
+          "Type"
+        ]
+      },
+      "then": {
+        "additionalProperties": false,
+        "properties": {
+          "Condition": {
+            "$ref": "file:common.json#definitions/Condition"
+          },
+          "DependsOn": {
+            "$ref": "file:common.json#definitions/DependsOn"
+          },
+          "Metadata": {
+            "type": "object"
+          },
+          "Properties": {
+            "additionalProperties": false,
+            "properties": {
+              "AccessLogSetting": {
+                "$ref": "file:cfn.json#definitions/AWS::ApiGateway::Stage.AccessLogSetting"
+              },
+              "Auth": {
+                "$ref": "#definitions/AWS::Serverless::Api.Auth"
+              },
+              "BinaryMediaTypes": {
+                "items": {
+                  "type": [
+                    "string",
+                    "intrinsic"
```

Comment:
General question on the type "intrinsic". How are we determining what can be intrinsic or not? Is this done by testing each field manually or are we including all fields that gets directly passed to CFN?

Reply:
All fields that we directly pass to CFN could receive an intrinsic.
We also resolve some intrinsic `Ref` so these should also be validated.
If we need the value (we transform a list of strings to a list of objects for example), we currently don't support intrinsic for these properties.